assets/js/hooks/useAdSenseNotifications.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * AdSense notifications side-effect hook.
+ *
+ * Site Kit by Google, Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { useDispatch, useSelect } from 'googlesitekit-data';
+import { CORE_MODULES } from '../googlesitekit/modules/datastore/constants';
+import { MODULES_ADSENSE } from '../modules/adsense/datastore/constants';
+import { MODULE_SLUG_ADSENSE } from '../modules/adsense/constants';
+import { NOTIFICATION_AREAS } from '../googlesitekit/notifications/constants';
+import { CORE_NOTIFICATIONS } from '../googlesitekit/notifications/datastore/constants';
+import NotificationFromServer from '../components/NotificationFromServer';
+
+export default function useAdSenseNotifications() {
+	const adSenseModuleConnected = useSelect( ( select ) =>
+		select( CORE_MODULES ).isModuleConnected( MODULE_SLUG_ADSENSE )
+	);
+	const accountID = useSelect( ( select ) =>
+		select( MODULES_ADSENSE ).getAccountID()
+	);
+
+	const [ registeredNotifications, setRegisteredNotifications ] = useState(
+		[]
+	);
+	const { registerNotification } = useDispatch( CORE_NOTIFICATIONS );
+
+	const notifications = useSelect( ( select ) =>
+		select( MODULES_ADSENSE ).getNotifications()
+	);
+
+	useEffect( () => {
+		// If AdSense is not connected, or the account ID is not set,
+		// there's nothing to do here.
+		if ( ! adSenseModuleConnected || ! accountID ) {
+			return;
+		}
+
+		// Register any notifications from the server that haven't yet been
+		// registered.
+		//
+		// (Usually there will be one, if any, notification from the server.)
+		notifications?.forEach( ( notification ) => {
+			if ( registeredNotifications.includes( notification.id ) ) {
+				return;
+			}
+
+			registerNotification( notification.id, {
+				Component() {
+					return <NotificationFromServer { ...notification } />;
+				},
+				priority: notification.priority,
+				areaSlug: NOTIFICATION_AREAS.DASHBOARD_TOP,
+				isDismissible: notification.isDismissible,
+			} );
+
+			setRegisteredNotifications( ( previousRegisteredNotifications ) => {
+				previousRegisteredNotifications.push( notification.id );
+
+				return previousRegisteredNotifications;
```

Comment:
Isn't this essentially mutating `registeredNotifications`?
I think this form of state update needs to be a pure function.
```suggestion
				return [ ...previousRegisteredNotifications, notification.id ];
```